wire-ios/Wire-iOS/Sources/UserInterface/Settings/ChangingDetails/ChangeEmailViewModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireSyncEngine
+
+final class ChangeEmailViewModel {
+
+    private(set) var state: ChangeEmailState
+    private weak var userProfile: UserProfile?
+
+    init(currentEmail: String?, userProfile: UserProfile?) {
+        self.state = ChangeEmailState(currentEmail: currentEmail ?? "")
+        self.userProfile = userProfile
+    }
+
+    func updateNewEmail(_ newEmail: String) {
+        state.newEmail = newEmail
+    }
+
+    func updateEmailValidationError(_ error: TextFieldValidator.ValidationError?) {
+        state.emailValidationError = error
+    }
+
+    func requestEmailUpdate(completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
Maybe I'm missing something, but I don't understand why this has a completion instead of being:

```swift
func requestEmailUpdate() throws
```

There doesn't seem to be anything asynchronous happening. Is it a kind of view model pattern?

Reply:
@samwyndham I made the updates you can find them here: https://github.com/wireapp/wire-ios/commit/529ccb786e17c4b45e695aedb61feef240a9d6e8

Reply:
Nice!